core/worker.go

```diff
@@ -1125,14 +1125,23 @@ func (w *worker) prepareWork(genParams *generateParams, wo *types.WorkObject) (*
 		if order == common.PRIME_CTX {
 			// Set the prime terminus
 			newWo.Header().SetPrimeTerminus(parent.Hash())
+			newWo.WorkObjectHeader().SetPrimeTerminusNumber(parent.Number(common.PRIME_CTX))
 		} else {
 			if w.hc.IsGenesisHash(parent.Hash()) {
 				newWo.Header().SetPrimeTerminus(parent.Hash())
+				newWo.WorkObjectHeader().SetPrimeTerminusNumber(big.NewInt(0))
 			} else {
 				// carry the prime terminus from the parent block
 				newWo.Header().SetPrimeTerminus(parent.Header().PrimeTerminus())
+				newWo.WorkObjectHeader().SetPrimeTerminusNumber(parent.WorkObjectHeader().PrimeTerminusNumber())
 			}
 		}
+	} else {
+		if w.hc.IsGenesisHash(parent.Hash()) {
+			newWo.WorkObjectHeader().SetPrimeTerminusNumber(big.NewInt(0))
```

Comment:
this is not always true, we need to inherit the parents number

Reply:
Done